#779: Unit tests for Wifi classes from core.net

djurod opened the pull request:
- Prepared unit tests for classes WifiAccessPointImpl,
  WifiInterfaceAddressConfigImpl and WifiInterfaceConfigImpl from
  package org.eclipse.kura.core.net
- Fixed or implemented comparison of classes: WifiAccessPointImpl,
  WifiInterfaceAddressConfigImpl

Signed-off-by: Djuro Drljaca <djuro.drljaca@comtrade.com>

amitjoy commented on kura/org.eclipse.kura.core.net/src/main/java/org/eclipse/kura/core/net/WifiAccessPointImpl.java:
```diff
@@ -141,4 +142,82 @@ public String toString() {
     }
+    
+    @Override
+	public int hashCode() {
```
To save verbosity, you could use `Objects.hash(Object...)` and `Objects.equals(Object, Object)`

djurod replied:
Could you explain what you mean exactly?

I generated this with the Eclipse IDE, and as far as I understand the equal method needs to be explicitly overridden, otherwise the comparison of the contents is not done (only references are compared)...

amitjoy replied:
Yeah, you are right. Equals must be consistent with hashcode to have it work with hash-based data structures. The IDE generated implementations of these `equals` and `hashcode` methods are verbose and expose low-level complexities. That's why in Java 7, few useful static factory methods are provided to write your own `equals` and `hashcode` implementation in a cleaner way. Have a look at the `hash(Object...)` and `equals(Object, Object)` methods in `Objects`.

djurod replied:
I checked this and also discussed it a little with @markoer, but in the end I came to a conclusion that the safest thing is to just generate this from Eclipse. It is less error prone than to manually create or update the methods. If any data member is changed you would just regenerate the methods using Eclipse...

amitjoy replied:
I don't have any problem with the implementation that Eclipse IDE generates for `equals` and `hashcode`. I personally prefer using`Objects` to write less verbose implementations. It's just a syntactic sugar to the code. But if you feel that it would be easier for you to manage this way, I am definitely okay with that 👍 .

amitjoy commented on kura/test/org.eclipse.kura.core.net.test/src/test/java/org/eclipse/kura/core/net/WifiAccessPointImplTest.java:
```diff
@@ -0,0 +1,497 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Eurotech and/or its affiliates and others
```
2017

djurod replied:
The code was prepared in 2016, but I can fix it :)

amitjoy replied:
So it would be then - `Copyright (c) 2016, 2017 Eurotech and/or its affiliates and others`

amitjoy commented on kura/test/org.eclipse.kura.core.net.test/src/test/java/org/eclipse/kura/core/net/WifiInterfaceAddressConfigImplTest.java:
```diff
@@ -0,0 +1,311 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Eurotech and/or its affiliates and others
```
2017

amitjoy commented on kura/test/org.eclipse.kura.core.net.test/src/test/java/org/eclipse/kura/core/net/WifiInterfaceConfigImplTest.java:
```diff
@@ -0,0 +1,156 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Eurotech and/or its affiliates and others
```
2017

amitjoy reviewed (changes requested):
Minor Changes

Commit:
#779: Unit tests for Wifi classes from core.net

- Prepared unit tests for classes WifiAccessPointImpl,
  WifiInterfaceAddressConfigImpl and WifiInterfaceConfigImpl from
  package org.eclipse.kura.core.net
- Fixed or implemented comparison of classes: WifiAccessPointImpl,
  WifiInterfaceAddressConfigImpl

Signed-off-by: Djuro Drljaca <djuro.drljaca@comtrade.com>
